### DISCUSSION TITLE
id in frontmatter works great in neovim, but Obsidian app is confused about wikilinks with id as path.

### DISCUSSION THREAD

**Discussion by joshfullmer** [epwalsh/obsidian.nvim#735] [Q&A]
I'm admittedly quite new to both using Obsidian and using the obsidian.nvim plugin, so I very well could be missing something (in fact, I hope that's the case). 

First, here's my full obsidian.nvim config: 

```lua
return {
  "epwalsh/obsidian.nvim",
  version = "*", -- recommended, use latest release instead of latest commit
  lazy = true,
  ft = "markdown",
  dependencies = {
    "nvim-lua/plenary.nvim",
  },
  config = function()
    require("obsidian").setup({
      workspaces = {
        {
          name = "personal",
          path = "~/vaults/personal",
        },
      },

      completion = {
        nvim_cmp = true,
        min_chars = 2,
      },

      follow_url_func = function(url)
        -- vim.fn.jobstart({ "open", url }) -- Mac OS
        vim.ui.open(url) -- need Neovim 0.10.0+
      end,

      ---@param spec { id: string, dir: obsidian.Path, title: string|? }
      ---@return string|obsidian.Path The full path to the new note.
      note_path_func = function(spec)
        -- This is equivalent to the default behavior.
        local path = spec.dir / tostring(spec.title)
        return path:with_suffix(".md")
      end,

      wiki_link_func = "prepend_note_path",

      daily_notes = {
        folder = "/2. Areas/Journal/",
      },
    })

    vim.keymap.set(
      "n",
      "<leader>oc",
      "<cmd>lua require('obsidian').util.toggle_checkbox()<CR>",
      { desc = "Obsidian Check Checkbox" }
    )
    vim.keymap.set("n", "<leader>ot", "<cmd>ObsidianTemplate<CR>", { desc = "[T]emplate" })
    vim.keymap.set("n", "<leader>oo", "<cmd>ObsidianOpen<CR>", { desc = "[O]pen in Obsidian App" })
    vim.keymap.set("n", "<leader>ob", "<cmd>ObsidianBacklinks<CR>", { desc = "[B]acklinks" })
    vim.keymap.set("n", "<leader>ol", "<cmd>ObsidianLinks<CR>", { desc = "[L]inks" })
    vim.keymap.set("n", "<leader>on", "<cmd>ObsidianNew<CR>", { desc = "[N]ew Note" })
    vim.keymap.set("n", "<leader>os", "<cmd>ObsidianSearch<CR>", { desc = "[S]earch Obsidian (grep)" })
    vim.keymap.set("n", "<leader>oq", "<cmd>ObsidianQuickSwitch<CR>", { desc = "[Q]uick Switch" })
    vim.keymap.set("n", "<leader>od", "<cmd>ObsidianToday<CR>", { desc = "[D]aily Note" })
  end,
}
```

Currently, my workflow is to create notes with `ObsidianNew` using the keymap. For example, I create a file `TODO`, the filename is `TODO.md`, and it adds frontmatter that looks like this:

```yaml
---
id: 1727199696-DZNB
aliases:
  - TODO
tags: []
---
```

Then, I'll use a "wikilink" in another file to link back to this file, like this:

```md
[[1727199696-DZNB|TODO]]
```

I can use `gf` or `<Enter>` with my cursor over that wikilink to successfully visit that file in neovim. However, the real problem I'm experiencing is in the Obsidian app. It's not recognizing the id frontmatter as a unique identifier of files. Clicking on any of those wikilinks in the app ends up creating a file with the id as the name. Furthermore, the graph is totally confused, having duplicates of id and the actual filename.

Ideally, I could just change some setting either in obsidian.nvim or in the app that would more easily support my flow. One thing that I would really like to maintain with my workflow is duplicate filenames in different directories, which is really why id is appealing, but I'm open to other solutions. 

Thanks for any help/ideas in advance!

**Comment by ghost** [comment]
Same issue here. My current workaround is deleting the `id` field in the frontmatter of my note, renaming it with `:ObsidianRename` and writing the file, so the `id` field is added back with a different value matching the new file name.

This works, but it's a bit tedious, plus it destroys the purpose of notes having unique IDs. Have you found a better solution?
